plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2081,6 +2092,9 @@ public ConnectorAnalyzeMetadata getStatisticsCollectionMetadata(ConnectorSession
                     alreadyAnalyzedModifiedTimeMax.orElse(EPOCH)));
         }
 
+        Set<String> allColumnNames = extractColumnMetadata(metadata, typeManager).stream()
+                .map(ColumnMetadata::getName)
+                .collect(toImmutableSet());
```

Comment:
Save the result of `extractColumnMetadata` so that you don't have to call it again at the bottom of this method.
```suggestion
        List<ColumnMetadata> columnMetadata = extractColumnMetadata(metadata, typeManager);
        Set<String> allColumnNames = columnMetadata.stream()
                .map(ColumnMetadata::getName)
                .collect(toImmutableSet());
```